ui/app/adapters/secret-v2-version.js

```diff
@@ -27,9 +28,9 @@ export default ApplicationAdapter.extend({
 
   findRecord() {
     return this._super(...arguments).catch(errorOrModel => {
-      // if it's a real 404, this will be an error, if not
-      // it will be the body of a deleted / destroyed version
-      if (errorOrModel instanceof DS.AdapterError) {
+      // if the response is a real 404 or if the secret is gated by a control group this will be an error,
+      // otherwise the response will be the body of a deleted / destroyed version
+      if (errorOrModel instanceof DS.AdapterError || errorOrModel instanceof ControlGroupError) {
```

Comment:
Idle food for thought: Does it make sense for `ControlGroupError` to extend `AdapterError`? If I recall correctly, this is how the other Ember Data errors (e.g., `NotFoundError`) work.

Reply:
hmm, good question. i don't see a problem with doing that, especially since `AdapterError` uses `EmberError` under the hood. thanks for the tip!